src/contrib/dependencyinjection/Akka.DependencyInjection/DependencyResolverSetup.cs

```diff
@@ -8,6 +8,23 @@
 using System;
 using Akka.Actor;
 using Akka.Actor.Setup;
+using Akka.DependencyInjection;
+
+namespace Akka.Actor
+{
+    public static class DependencyInjectionSetupExtensions
+    {
+        public static BootstrapSetup WithDependencyInjection(this BootstrapSetup bootstrap, IServiceProvider provider)
+        {
+            return bootstrap.WithExtensions(new DependencyResolverExtension(provider));
```

Comment:
I'd just combine setups here and return an `ActorSystemSetup` - this is what we do with Phobos, for instance: https://phobos.petabridge.com/articles/setup/configuration.html#phobossetup-and-phobosconfigbuilder

```csharp
var phobosSetup = PhobosSetup.Create(phobosConfigBuilder)
    .And(BootstrapSetup.Create()
        .WithConfig(hocon)
        .WithActorRefProvider(PhobosProviderSelection.Cluster));
```

The `PhobosSetup.Create` method actually return an `ActorSystemSetup` because it combines both a `TracingSetup` and a separate `MonitoringSetup` type internally.

Reply:
Using that type of composition is preferable to changing the `DependencyResolverExtension` - IMHO.